### PR TITLE
🚀 Improve code resilience

### PR DESCRIPTION
**Pull request by n3a9**
**Rate limiting for authentication**
For all authentication routes, add [rate limiting](https://codeql.github.com/codeql-query-help/javascript/js-missing-rate-limiting/).

**Explicit null checks**
Avoid `== null` and `!= null` checks as they have much more [complicated behavior](https://262.ecma-international.org/5.1/#sec-11.9.3). Explicit checks also make it clear for code reader for what is being checked.

**Prefer nullish coalescing operator**
Prefer `string ?? ''` to `string || ''` as nullish coalescing operator is [safer and more predictable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator#assigning_a_default_value_to_a_variable).

### CODE REVIEW

File: client/src/components/desktop/ResourceDetail.jsx [hunk 1]
```diff
@@ -84,51 +84,53 @@ function ResourceDetail(props) {
...
-        setFinancialAidDetails(result.financialAidDetails);
-        setContacts(result.contacts);
+        setRequiredDocuments(res.requiredDocuments);
+        setFinancialAidDetails(res.financialAidDetails);
+        setContacts(res.contacts);
 
         if (authed) {
           let savedSet = new Set();
           const json = await getSavedResources();
-          savedSet = new Set(json.result);
```
Comment (jackieo5023) [inline client/src/components/desktop/ResourceDetail.jsx:131]: Shouldn't this still be result?

File: client/src/components/desktop/ResourceDetail.jsx [hunk 2]
```diff
@@ -144,7 +146,7 @@ function ResourceDetail(props) {
       if (authed) {
         let savedSet = new Set();
         const json = await getSavedResources();
-        savedSet = new Set(json.result);
```
Comment (jackieo5023) [inline client/src/components/desktop/ResourceDetail.jsx:147]: And this

File: client/src/components/desktop/ResourceDetail.jsx [hunk 3]
```diff
@@ -156,7 +158,7 @@ function ResourceDetail(props) {
       const language = localStorage.getItem('language');
       if (language !== 'English') {
         const response = await getResourceIsVerified(match.params.id, language);
-        setIsVerified(response.result);
```
Comment (jackieo5023) [inline client/src/components/desktop/ResourceDetail.jsx:159]: And this

File: client/src/components/mobile/ResourceDetailMobile.jsx [hunk 4]
```diff
@@ -154,7 +158,7 @@ const ResourceDetailMobile = (props: Props) => {
       const language = localStorage.getItem('language');
       if (language !== 'English') {
         const response = await getResourceIsVerified(resourceId, language);
-        setIsVerified(response.result);
+        setIsVerified(response.res);
```
Comment (jackieo5023) [inline client/src/components/mobile/ResourceDetailMobile.jsx:161]: And this

File: client/src/components/mobile/ResourceDetailMobile.jsx [hunk 5]
```diff
@@ -181,7 +185,7 @@ const ResourceDetailMobile = (props: Props) => {
       if (authed) {
         let savedSet = new Set();
         const json = await getSavedResources();
-        savedSet = new Set(json.result);
```
Comment (jackieo5023) [inline client/src/components/mobile/ResourceDetailMobile.jsx:184]: And this

File: api/src/scripts/addVerifiedTranslations.js [hunk 6]
```diff
@@ -124,18 +128,20 @@ async function main() {
...
       }
-      if (requiredDocuments != null && requiredDocuments.length !== 0) {
-        let idx = 0;
-        for (const requiredDoc of requiredDocuments) {
+      if (
+        requiredDocuments !== null &&
+        requiredDocuments !== undefined &&
+        requiredDocuments.length > 0
+      ) {
+        for (let i = 0; i < requiredDocuments.length; i++) {
```
Comment (jackieo5023) [inline api/src/scripts/addVerifiedTranslations.js:136]: nice LOL idk what I was doing there

**Review by jackieo5023 — APPROVED** [review]
awesome thanks :)

File: auth/src/App.js [hunk 7]
```diff
@@ -4,17 +4,24 @@ const bodyParser = require("body-parser");
...
+const RateLimit = require("express-rate-limit");
 
 const app = express();
 require("dotenv").config();
 
+const limiter = new RateLimit({
+  windowMs: 1 * 60 * 1000,
+  max: 5,
+});
+
```
Comment (jackieo5023) [inline auth/src/App.js:16]: What's the reasoning behind 5 requests per minute? I'm just trying to think if there's a case we'd go over that. Like login, verify, get saved resources? I'm not sure how often we call verify

File: client/src/components/desktop/ResourceDetail.jsx (hunk 1 above)
Comment (n3a9) [reply client/src/components/desktop/ResourceDetail.jsx:131]: oh ya, oops. just did search / replace for these :P

File: auth/src/App.js (hunk 7 above)
Comment (n3a9) [reply auth/src/App.js:16]: Just the default that the documentation provided. I think it's pretty suitable, not sure how much we need to play around with it unless you think we should increase it

**Pull request merged**